### PR TITLE
Promote `UseEtcdWrapper` feature gate to GA

### PR DESCRIPTION
**Pull request by shreyas-s-rao**
**How to categorize this PR?**

/area quality open-source usability
/kind task

**What this PR does / why we need it**:
The feature gate `UseEtcdWrapper` has now been promoted to GA, and locked to true. The related code has been cleaned up. The feature gate will finally be removed in #935 .

**Which issue(s) this PR fixes**:
Fixes partially #776 

**Special notes for your reviewer**:
/assign @seshachalam-yv @ishan16696 
/cc @unmarshall @aaronfern 

**Release note**:

```noteworthy user
The `UseEtcdWrapper` feature gate has been promoted to GA and locked to `true`.
```

### CODE REVIEW

**Comment by shreyas-s-rao** [comment]
I have also raised https://github.com/gardener/ci-infra/pull/2757 to remove the prow job for e2e test for nondistroless version of etcd (ie, etcd-custom-image), since that is no longer supported now that the feature gate has been promoted to GA.

**Comment by ishan16696** [comment]
@shreyas-s-rao please rebase this PR as `etcd-backup-restore` version has been updated

**Comment by shreyas-s-rao** [comment]
@ishan16696 rebased as requested. Although, I would want to wait for #812 to be merged before rebasing this PR again and then merging. I don't want @anveshreddy18 to go through yet another round of rebasing on his PR. Until I rebase on the configmap changes, the tests here will continue to fail, so these test failures can be ignored for the timebeing.

**Review by seshachalam-yv — APPROVED** [review]
Thanks @shreyas-s-rao for promoting the feature gate

**Review by seshachalam-yv — APPROVED** [review]

**Pull request merged**
